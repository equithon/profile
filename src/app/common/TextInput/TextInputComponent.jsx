import React, { useState, useEffect } from "react";
import styled from "styled-components";

const Container = styled.input`
  width: 100%;
  height: 100%;
  padding: 0.8em;

  font-size: 100%;
  color: ${props =>
    props.theme.colors[props.outlineColor] ||
    props.outlineColor ||
    props.theme.colors.black};

  border-radius: ${props => props.theme.app.border.radius};
  border: ${props =>
    `2px solid ${props.theme.colors[props.outlineColor] ||
      props.outlineColor ||
      props.theme.colors.lightBlack}`};

  transition: box-shadow 400ms ease-in-out;
  &.touched,
  &:focus {
    outline: none;
    box-shadow: ${props =>
      `0 0 3px 1px ${props.theme.colors[props.outlineColor] ||
        props.outlineColor ||
        props.theme.colors.lightBlack}`};
  }
`;

const TextInput = ({
  className,
  name,
  type = "text",
  defaultValue = "",
  placeholder,
  outlineColor,
  onChangeHandler = () => {},
  onBlurHandler = () => {},
  formikInfo = { field: {}, form: {} }
}) => {
  const [curValue, updateCurValue] = useState(defaultValue);

  useEffect(() => {
    if (!formikInfo) onChangeHandler(curValue);
  });

  return (
    <Container
      className={className}
      name={name}
      type={type}
      placeholder={placeholder}
      outlineColor={outlineColor}
      onChange={e => updateCurValue(e.target.value)}
      onBlur={onBlurHandler}
      {...formikInfo.field}
      formikForm={formikInfo.form}
    />
  );
};

export default TextInput;
