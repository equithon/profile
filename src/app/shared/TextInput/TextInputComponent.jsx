import React, { useState, useEffect } from "react";
import styled from "styled-components";

const TextInput = styled.input`
  width: 100%;
  height: 100%;
  padding: ${props => props.theme.app.container.padding};

  font-size: 100%;
  color: ${props =>
    props.theme.colors[props.outlineColor] || props.outlineColor};

  border-radius: ${props => props.theme.app.border.radius};
  border: ${props =>
    `2px solid ${props.theme.colors[props.outlineColor] ||
      props.outlineColor}`};

  transition: box-shadow 400ms ease-in-out;
  &:focus {
    outline: none;
    box-shadow: ${props =>
      `0 0 3px 1px ${props.theme.colors[props.outlineColor] ||
        props.outlineColor}`};
  }
`;

export default ({
  className,
  name,
  type = "text",
  defaultValue = "",
  placeholder = `Enter ${type} here`,
  outlineColor = "lightBlack",
  onChangeHandler = () => {},
  onBlurHandler = () => {},
  formikInfo = { notUsing: true, field: {}, form: {} }
}) => {
  const [value, updateValue] = useState(defaultValue);

  useEffect(() => {
    if (formikInfo.notUsing) onChangeHandler(value);
  });

  return (
    <TextInput
      className={className}
      outlineColor={outlineColor}
      name={name}
      type={type}
      placeholder={placeholder}
      onChange={e => updateValue(e.target.value)}
      onBlur={onBlurHandler}
      {...formikInfo.field}
      formikForm={formikInfo.form}
    />
  );
};
