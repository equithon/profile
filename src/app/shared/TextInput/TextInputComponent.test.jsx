/* eslint-disable no-undef */
import React from "react";

import TextInput from "./TextInputComponent";

describe("TextInput", () => {
  it("renders correctly", () => {
    const component = mountWithTheme(
      <TextInput placeholder="This is a test text area" />
    );
    expect(component).toMatchSnapshot();
  });
});
