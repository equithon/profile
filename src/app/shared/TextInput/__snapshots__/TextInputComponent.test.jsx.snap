// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 0.8em;
  font-size: 100%;
  color: #5c5c5c;
  border-radius: 6px;
  border: 2px solid #5c5c5c;
  -webkit-transition: box-shadow 400ms ease-in-out;
  transition: box-shadow 400ms ease-in-out;
}

.c0:focus {
  outline: none;
  box-shadow: 0 0 3px 1px #5c5c5c;
}

.c0:-moz-read-only {
  background-color: hsl(0,0%,95%);
  color: #aaaaaa;
}

.c0:read-only {
  background-color: hsl(0,0%,95%);
  color: #aaaaaa;
}

<_default
  placeholder="This is a test text area"
>
  <styled.input
    formikForm={Object {}}
    onBlur={[Function]}
    onChange={[Function]}
    outlineColor="lightBlack"
    placeholder="This is a test text area"
    type="text"
    value=""
  >
    <StyledComponent
      formikForm={Object {}}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 100%;
  height: 100%;
  padding: ",
              [Function],
              ";

  font-size: 100%;
  color: ",
              [Function],
              ";

  border-radius: ",
              [Function],
              ";
  border: ",
              [Function],
              ";

  transition: box-shadow 400ms ease-in-out;
  &:focus {
    outline: none;
    box-shadow: ",
              [Function],
              ";
  }

  &:read-only {
    background-color: hsl(0,0%,95%);
    color: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onBlur={[Function]}
      onChange={[Function]}
      outlineColor="lightBlack"
      placeholder="This is a test text area"
      type="text"
      value=""
    >
      <input
        className="c0"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="This is a test text area"
        type="text"
        value=""
      />
    </StyledComponent>
  </styled.input>
</_default>
`;
